### PR TITLE
fix(Scripts/Spells): Moonkin Form passive aura should not proc off fr…

### LINKED ISSUE #12435: [Druid] Moonkin mana restore procs on aoe
### What client do you play on?

enUS

### Faction

- [X] Alliance
- [X] Horde

### Content Phase:

- [X] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59
- [ ] 60

### Current Behaviour

Moonkin form reads 

> Single target spell critical strikes in this form instantly regenerate 2% of your total mana.  The Moonkin can not cast healing or resurrection spells while shapeshifted.

This procs on Hurricane and Starfall crits

image

### Expected Blizzlike Behaviour

Only procs on single target crits

### Source

Single target is not aoe

### Steps to reproduce the problem

Be a druid
Go into moonkin form (get lots of crit, forgot the name of the debug rings)
Cast Starfall/Hurricane
Look at your mana/combatlog

### Extra Notes

https://wowgaming.altervista.org/aowow/?spell=24858
https://wowgaming.altervista.org/aowow/?spell=48467
https://wowgaming.altervista.org/aowow/?spell=48505
Originally reported: https://github.com/chromiecraft/chromiecraft/issues/3774
### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/35dfcf588f62339c44ff99a3e7e85f0b983ed923

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-bg-item-reward](https://github.com/azerothcore/mod-bg-item-reward)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-cta-switch](https://github.com/azerothcore/mod-cta-switch)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-dmf-switch](https://github.com/azerothcore/mod-dmf-switch)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna](https://github.com/azerothcore/mod-eluna)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-low-level-rbg](https://github.com/azerothcore/mod-low-level-rbg)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-progression-system](https://github.com/azerothcore/mod-progression-system)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-rdf-expansion](https://github.com/azerothcore/mod-rdf-expansion)
- [mod-transmog](https://github.com/azerothcore/mod-transmog)
- [mod-weekend-xp](https://github.com/azerothcore/mod-weekend-xp)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations

None

### Server

ChromieCraft

### PR DESCRIPTION
**Pull request by UltraNix**
…om AoE spells.

Fixes #12435

## Issues Addressed:

- Closes #12435
- Closes https://github.com/chromiecraft/chromiecraft/issues/3774

## Tests Performed:

- Not tested.

## How to Test the Changes:

Be a druid
Go into moonkin form (get lots of crit, forgot the name of the debug rings)
Cast Starfall/Hurricane
Look at your mana/combatlog

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Gultask** [comment]
Doesn't seem to work.
image

**Review by IntelligentQuantum — APPROVED** [review]

**Comment by Gultask** [comment]
Tested with Starfall and Hurricane, both are still proccing power gain on crits.

`.learn 24858` `.learn 48505` `.learn 16914`
`.go xyz 2107 -4667 45.801258 1 5.5`

**Comment by UltraNix** [comment]
Just tested - works okay.
obraz_2022-08-27_163718641

File: src/server/scripts/Spells/spell_druid.cpp [hunk 1]
```diff
@@ -1150,6 +1150,27 @@ class spell_dru_berserk : public SpellScript
     }
 };
 
+// 24905 - Moonkin Form (Passive) (Passive)
```
Comment (Nefertumm) [inline src/server/scripts/Spells/spell_druid.cpp:1153]:
```suggestion
// 24905 - Moonkin Form (Passive)
```

**Pull request merged**